src/LinReg_LA_adapt.cpp

```diff
@@ -32,61 +32,59 @@ namespace LinReg_LA_adapt {
 using namespace std;
 using namespace LinReg_LA_adapt;
 
-// LinRegLA_adapt_impl() function callable from R via Rcpp:: 
+// LinRegLA_adapt_impl() function callable from R via Rcpp::
 // [[Rcpp::export]]
-Rcpp::List LinRegLA_adapt_impl(arma::mat Data, unsigned long lNumber, double resampTol, double tempTol) {     
+Rcpp::List LinRegLA_adapt_impl(arma::mat Data, unsigned long lNumber, double resampTol, double tempTol) {
 
 
     try {
         rho = tempTol;
-        
+
         lIterates = Data.n_rows;
         data.y = Data.col(0);
         data.x = Data.col(1);
         mean_x = arma::sum(data.x)/lIterates;
-        
+
         // Initialise the sampler
         myAdapt = new rad_adapt;
         Sampler = new smc::sampler<rad_state,smc::staticModelAdapt>(lNumber, HistoryType::RAM);
         smc::moveset<rad_state,smc::staticModelAdapt> Moveset(fInitialise, fMove, fMCMC);
-        
+
         Sampler->SetResampleParams(ResampleType::SYSTEMATIC, resampTol);
         Sampler->SetMoveSet(Moveset);
         Sampler->SetAlgParam(smc::staticModelAdapt());
         Sampler->SetAdaptMethods(myAdapt);
         Sampler->Initialise();
-        
+
         //Run the sampler
         while (Sampler->GetAlgParams().GetTempCurr() != 1){
             Sampler->Iterate();
         }
-        
+
         //Storing the results in a sensible format
         std::vector<double> temps = Sampler->GetAlgParams().GetTemps();
         int lTemps = temps.size();
-        
-        std::vector<smc::historyelement<rad_state> > Hist = Sampler->GetHistory();
-        
+
         arma::cube theta(lNumber,3,lTemps);
         arma::mat loglike(lNumber,lTemps), logprior(lNumber,lTemps), Weights(lNumber,lTemps);
         arma::vec ESS(lTemps), mcmcRepeats(lTemps);
-        
+
         for(int n=0; n < lTemps; ++n) {
             for (unsigned int i=0; i<lNumber; i++){
-                theta.slice(n).row(i) = Hist[n].GetValues().GetValueN(i).theta.t();
-                loglike(i,n) = Hist[n].GetValues().GetValueN(i).loglike;
-                logprior(i,n) = Hist[n].GetValues().GetValueN(i).logprior;
+                theta.slice(n).row(i) = Sampler->GetHistoryPopulation(n).GetValueN(i).theta.t();
+                loglike(i,n) = Sampler->GetHistoryPopulation(n).GetValueN(i).loglike;
```

Comment:
This is a general thought not directly related to the undoubted usefulness of the current PR, but if we envisage users wanting to access a particular value from a particular iteration on a regular basis then perhaps a `GetHistoryValue(n,i)` helper which is equivalent to `GetHistoryPopulation(n).GetValueN(i)` and a corresponding `GetHistoryWeight(n,i)` might be worth adding. Any thoughts?

Reply:
I think it may be harder for the users to use it. I have worked at the nautilus project and I think that i have learned from them that we need to stop with the OOP when the documentation necessary to understand the code is longer than the previous code itself. But i would like to get the opinion of a more seasoned member of the community.

Reply:
I would imagine having two getters like `GetValueN(i)` and `GetValueN(i,t)` would be sensible. I think it would be nice for users to be able to access as many things as possible directly through the sampler, but I don't have any particularly strong preferences.

Reply:
More code and more documentation isn't always good, for sure, I have a vague feeling this might be a useful enough feature to warrant adding, but it doesn't block merging what is already proposed here.

Reply:
I will make  another commit with this changes later. I think that we may exaggerate with the changes if I include this in the commit as well. And also the meter is a bit disputable.